### PR TITLE
ENH: workflows update and new docker files

### PR DESCRIPTION
**Pull request by Gui-FernandesBR**
## Pull request type

- [ ] Code changes (bugfix, features)
- [ ] Code maintenance (refactoring, formatting, tests)
- [x] ReadMe, Docs and GitHub updates
- [ ] Other (please describe):

## Checklist

- [ ] Tests for the changes have been added (if needed)
- [x] Docs have been reviewed and added / updated
- [x] Lint (`black rocketpy/ tests/`) has passed locally 
- [ ] All tests (`pytest --runslow`) have passed locally

## Current behavior
- Our github actions only support python 3.11, but we should be already testing the new python 3.12.
- It is hard to test if rocketpy is running in all the 3 supported OS before submitting a PR.

## New behavior
- Docker is here! I added a `Dockerfile` and a `docker-compose.yml` file to build and run rocketpy in Docker containers.
- Some short instructions on how to use Docker were documented.  
- GitHub actions are still the most reliable and easy way to test rocketpy on different OS, but docker allows us to do it out of GitHub now, free of charge.
- Travis CI was disabled, and its file deleted. We cannot use that workflow due to the free plan constrain. Also, it started to be a bit redundant since the last improvement of GitHub Actions.

## Breaking change

- [x] No

## Additional information

- More complex codes can be done with Docker in this project, but I think this PR is already a good start.
- RocketPy is building on Python 3.12, but the optional package `timezonefinder` can not be installed yet (see https://github.com/jannikmi/timezonefinder/issues/208)
- TLDR: if you code on windows, test your code on linux using docker! (vice-versa)

### CODE REVIEW

**Comment by Gui-FernandesBR** [comment]
it would be nice if we fix (or simply delete) the action that makes actions fail if the PR is marked as draft. 
It is not working, and sometimes it runs for hours because it never get closed. See an example: https://github.com/RocketPy-Team/RocketPy/actions/runs/6681843992

What is the point of not applying tests on the draft PRs?
- Draft PRs should not stay open for so long, 
- Have we ever get close to the github monthly quota limit?

**Comment by Gui-FernandesBR** [comment]
Example of project that also uses Dockerfile: https://github.com/pandas-dev/pandas/blob/main/Dockerfile

Thanks @phmbressan for mentioning this today.

**Comment by ringsaturn** [comment]
FYI: you can try https://github.com/ringsaturn/tzfpy under Python 3.12 before timezonefinder is ready for it.

**Comment by Gui-FernandesBR** [comment]
Tests are passing on Python 3.12 (good news!!).

Several deprecation warnings are being raised tho, let's take a look at those in the next weeks.

**Comment by Gui-FernandesBR** [comment]
> FYI: you can try https://github.com/ringsaturn/tzfpy under Python 3.12 before timezonefinder is ready for it.

Thank you for your suggestion, your project is very nice. We will take a deeper look in the future.

**Review by phmbressan — APPROVED** [review]
